site/en/integrations/integrate_with_voyageai.md

```diff
@@ -0,0 +1,149 @@
+---
+id: integrate_with_voyageai.md
+summary: This page discusses vector database integration with VoyageAI's embedding API.
+---
+
+# Similarity Search with Milvus and VoyageAI
+
+This page discusses vector database integration with VoyageAI's embedding API.
+
+We'll showcase how [VoyageAI's Embedding API](https://docs.voyageai.com/docs/embeddings) can be used with our vector database to search across book titles. Many existing book search solutions (such as those used by public libraries, for example) rely on keyword matching rather than a semantic understanding of what the title is actually about. Using a trained model to represent the input data is known as _semantic search_, and can be expanded to a variety of different text-based use cases, including anomaly detection and document search.
+
+## Getting started
+
+The only prerequisite you'll need here is an API key from the [VoyageAI website](https://dash.voyageai.com/api-keys). Be sure you have already [started up a Milvus instance](https://milvus.io/docs/install_standalone-docker.md).
+
+We'll also prepare the data that we're going to use for this example. You can grab the book titles [here](https://www.kaggle.com/datasets/jealousleopard/goodreadsbooks). Let's create a function to load book titles from our CSV.
+
+```python
+import csv
+import random
+import voyageai
+from pymilvus import connections, FieldSchema, CollectionSchema, DataType, Collection, utility
+```
+
+```python
+# Extract the book titles
+def csv_load(file):
+    with open(file, newline='') as f:
+        reader=csv.reader(f, delimiter=',')
+        for row in reader:
+            yield row[1]
+```
+
+With this, we're ready to move on to generating embeddings.
+
+## Searching book titles with VoyageAI & Milvus
+
+Here we can find the main parameters that need to be modified for running with your own accounts. Beside each is a description of what it is.
+
+```python
+FILE = './content/books.csv'  # Download it from https://www.kaggle.com/datasets/jealousleopard/goodreadsbooks and save it in the folder that holds your script.
+COLLECTION_NAME = 'title_db'  # Collection name
+DIMENSION = 1024  # Embeddings size
+COUNT = 100  # How many titles to embed and insert.
+MILVUS_HOST = 'localhost'  # Milvus server URI
+MILVUS_PORT = '19530'
+MODEL_NAME = 'voyage-2'  # Which engine to use, please check https://docs.voyageai.com/docs/embeddings for available models
+client = voyageai.Client(api_key="USE YOUR API KEY HERE")
+```
+
+This segment deals with Milvus and setting up the database for this use case. Within Milvus, we need to set up a collection and index the collection. For more information on how to use Milvus, look [here](https://milvus.io/docs/example_code.md).
+
+```python
+# Connect to Milvus
+connections.connect(host=MILVUS_HOST, port=MILVUS_PORT)
```

Comment:
We have recently changed the API to a more friendly `MilvusClient` interface. Let's keep the code here as is, and I'll draft a follow up PR to fix the code.